src/librustc_resolve/lib.rs

```diff
@@ -4783,10 +4774,17 @@ impl<'a, 'crateloader: 'a> Resolver<'a, 'crateloader> {
             };
 
             let cm = self.session.source_map();
-            let rename_msg = "You can use `as` to change the binding name of the import";
-
-            if let (Ok(snippet), false) = (cm.span_to_snippet(binding.span),
-                                           binding.is_renamed_extern_crate()) {
+            let rename_msg = "you can use `as` to change the binding name of the import";
+
+            if let (
+                Ok(snippet),
+                NameBindingKind::Import { directive, ..},
+                _x @ 1 ... std::u32::MAX,
+            ) = (
+                cm.span_to_snippet(binding.span),
+                binding.kind.clone(),
+                binding.span.hi().0,
```

Comment:
I believe you can have the same effect here by comparing `DUMMY_SP != binding.span`.

Reply:
I found `binding.span.is_dummy()` which looks good for that check